goplay/src/main/java/be/tapped/goplay/model.kt

```diff
@@ -1,42 +1,46 @@
 package be.tapped.goplay
 
+import arrow.core.Nel
 import be.tapped.goplay.content.Program
 import be.tapped.goplay.epg.EpgProgram
 import be.tapped.goplay.profile.TokenWrapper
 import kotlinx.datetime.LocalDate
 
-public sealed class ApiResponse {
-    public sealed class Success : ApiResponse() {
-        public sealed class Authentication : Success() {
-            public data class Token(val token: TokenWrapper) : Authentication()
-            public data class Profile(val profile: be.tapped.goplay.profile.Profile) : Authentication()
+public sealed interface ApiResponse {
```

Comment:
Do we need to unite `Success` and `Failure` under a single type?
Do we really need thee `Success` type?

Can we not get away with using `Either` to model this relationship?

Reply:
This would also shorten a lot of the return types.

Reply:
There are some duplicates names involved I think